nntrainer/graph/network_graph.cpp

```diff
@@ -457,6 +457,53 @@ void NetworkGraph::setBatchSize(unsigned int batch_size) {
     allocateTensors(exec_mode);
 }
 
+void NetworkGraph::tryZeroGrad(LayerNode *node) {
+  auto &rc = node->getRunContext();
+  auto current_execution_order = std::get<1>(node->getExecutionOrder());
+  auto num_weight = rc.getNumWeights();
+  for (unsigned i = 0; i < num_weight; ++i) {
+    if (!rc.weightHasGradient(i)) {
+      continue;
+    }
+
+    auto &grad = rc.getWeightGrad(i);
+    if (tensor_manager->isFirstAccess(grad.getName(),
+                                      current_execution_order)) {
+      grad.initialize();
+    }
+  }
+
+  auto num_input = rc.getNumInputs();
+  for (unsigned i = 0; i < num_input; ++i) {
+    auto deriv = rc.getOutgoingDerivative(i);
+    if (tensor_manager->isFirstAccess(deriv.getName(),
+                                      current_execution_order)) {
+      deriv.initialize();
```

Comment:
I also wonder if this is needed?

Reply:
The general behavior is to overwrite the gradient memory. This is being changed for gradients which has shared weights, but not for derivatives. Let me know if I am missing something.

Reply:
I think it's correct :)

Reply:
Deleted